src/controllers/mpris.js

```diff
@@ -0,0 +1,112 @@
+            })
+            Events.subscribe(Events.player.shuffle_changed, function(){
+                electron.ipcRenderer.send('readDZShuffle', dzPlayer.shuffle)
+            })
+            Events.subscribe(Events.player.repeat_changed, function(){
+                electron.ipcRenderer.send('readDZRepeat', dzPlayer.getRepeat())
+            })
+      `);
+
+        // The function used to know when to read the Deezer track position
+        this.player.getPosition = () => {
+            this.win.webContents.executeJavaScript(`
+            var electron = require('electron')
+            var value = dzPlayer.getPosition()
+            electron.ipcRenderer.send('readDZCurPosition', value)`);
+        };
+
+        this.player.on('quit', () => {
+            process.exit();
+        });
+    }
+
+    bindEvents() {
+        // MPRIS side actions
+        this.player.on('pause', () => {
+            this.win.webContents.executeJavaScript("dzPlayer.control.pause();");
+        })
+        this.player.on('play', () => {
+            this.win.webContents.executeJavaScript("dzPlayer.control.play();");
+        })
+        this.player.on('playpause', () => {
+            this.win.webContents.executeJavaScript("dzPlayer.control.togglePause();");
+        })
+        this.player.on('loopStatus', () => {
+            let repeat_status;
+            switch (this.player.loopStatus) {
+                case "None":
+                    this.player.loopStatus = "Playlist";
+                    repeat_status = 1;
+                    break;
+                case "Playlist":
+                    this.player.loopStatus = "Track";
+                    repeat_status = 2;
+                    break;
+                case "Track":
+                    this.player.loopStatus = "None";
+                    repeat_status = 0;
+                    break;
+            };
+            this.win.webContents.executeJavaScript(`dzPlayer.control.setRepeat(${repeat_status});`);
+        })
+        this.player.on('shuffle', () => {
+            let shuffle_status = arguments['0'];
+            this.player.shuffle = shuffle_status;
+            this.win.webContents.executeJavaScript(`dzPlayer.control.setShuffle(${shuffle_status});`);
+        })
+        this.player.on('next', () => {
+            this.win.webContents.executeJavaScript("dzPlayer.control.nextSong();");
+        })
+        this.player.on('previous', () => {
+            this.win.webContents.executeJavaScript("dzPlayer.control.prevSong();");
+        })
+        this.player.on('volume', () => {
+            this.win.webContents.executeJavaScript(`dzPlayer.control.setVolume(${arguments['0']});`);
+        })
+        this.player.on('position', () => {
+            let cur_pos = arguments['0']['position'];
+            let length = this.player.metadata['mpris:length'];
+            this.win.webContents.executeJavaScript(`dzPlayer.control.seek(${cur_pos / length});`);
+        })
```

Comment:
The seeking from plasma do not work actually. I dot have time to take a closer look at why, but I let you know.

Reply:
Thanks! Will take a look in the next few days

Reply:
https://github.com/fminetto/unofficial-deezer-player/pull/23/commits/c8e55653953c816facbf9e56e4348a691b22fead fixes setting the position, both playerctl position 10 and playerctl position 10+ work now.

Another problem I haven't fixed yet is reading the position via playerctl position 

`this.player.getPosition = () => {`
`            this.win.webContents.executeJavaScript("`
`            var electron = require('electron')`
`            var value = dzPlayer.getPosition()`
`            electron.ipcRenderer.send('readDZCurPosition', value)");`
`           // here we should return the read value, not just set it`
`};`

Sounds easy enough, although because dzPlayer is in another thread, it's hard to actually return it's value without blocking either of the threads, also would prefer solution without polling/setTimeout

Reply:
Probably fixed by https://github.com/fminetto/unofficial-deezer-player/pull/23/commits/83373acb6eb48abece7f95fd66593854e52951f6 it's not perfect solution, but it's a working one. Still would appreciate better approach.